hw/ip/spi_host/data/spi_host.hjson

```diff
@@ -0,0 +1,590 @@
+                   over-estimate, Due to synchronization delays,
+                ''',
+          resval: "0x0"
+        }
+      ]
+    },
+    { name: "CONFIG",
+      desc: '''Configuration register.  Contains fields for enabling the IP
+               .''',
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "29",
+          name: "CPOL",
+          desc: '''The polarity of the sck clock signal.  When CPOL is 0,
+                   sck is low when idle, and emits high pulses.   When CPOL
+                   is low, sck is high when idle, and emits a series of low
+                   pulses.
+                   '''
+          resval: "0x0"
+        },
+        { bits: "28",
+          name: "FULLCYC",
+          desc: '''Full cycle.  Modifies the CPHA sampling behaviour to allow
+                   for longer device logic setup times.  Rather than sampling the SD
+                   bus a half cycle after shifting out data, the data is sampled
+                   a full cycle after shifting data out.  This means that if
+                   CPHA = 0 data is both shifted the trailing edge, and sampled
+                   a full cycle later.  If CPHA = 1 data is shifted and
+                   sampled with the trailing edge, also separated by a
+                   full cycle.'''
+          resval: 0
+        },
+        { bits: "27",
+          name: "CPHA",
+          desc: '''The phase of the sck clock signal relative to the data. When
+                   CPHA = 0, the data changes on the trailing edge of sck
+                   and is typically sampled on the leading edge.  Conversely
+                   if CPHA = 1 high, data lines change on the leading edge of
+                   sck and are typically sampled on the trailing edge.
+                   CPHA should be chosen to match the phase of the selected
+                   device.  The sampling behavior is modified by the
+                   FLCYC bit.
+                   ''',
+          resval: "0x0"
+        },
+        { bits: "26",
+          name: "CSAAT",
+          desc: '''Chip select active after transaction.  If CSAAT = 0, the
+                   chip select line is raised immediately at the end of the
+                   transaction.   If CSAAT = 1, the chip select line is left
+                   low at the end of the current sequence.  This potentially
+                   allows for the creation of longer or otherwise more
+                   complex sequences.'''
+          resval: "0x0"
+        },
+        { bits: "25:23",
+          name: "CSNLEAD",
+          desc: '''CS_N Leading Time.  Indicates the number of half sck cycles,
+                   CSNLEAD+1, to leave between the falling edge of cs_n and
+                   the first edge of sck.  Setting this register to zero
+                   corresponds to the minimum delay of one-half sck cycle'''
+          resval: 0
+        },
+        { bits: "22:20",
+          name: "CSNTRAIL"
```

Comment:
I love the name! CS Entrail!

Reply:
Hmm.  Didn't think of that... Sadly I can't entirely take credit for the Lead, Trail, & Idle names.